query-engine/request-handlers/src/handler.rs

```diff
@@ -171,9 +172,13 @@ impl<'a> RequestHandler<'a> {
 
                                 responses.insert_data(&singular_name, Item::Map(result));
                             }
-                            _ => {
-                                responses.insert_data(&singular_name, Item::null());
-                            }
+                            None if throw_on_empty => responses.insert_error(GQLError::from_user_facing_error(
+                                user_facing_errors::query_engine::RecordRequiredButNotFound {
+                                    cause: "Expected a record, found none.".to_owned(),
+                                }
+                                .into(),
+                            )),
```

Comment:
A bit hacky but that should do the work 👍

Reply:
Indeed hacky, I tried to find some refactoring opportunity to consolidate this logic and the logic in the query interpreter that renders *OrThrow errors, but found it less clear than actually repeating myself.